src/test/java/com/zaxxer/nuprocess/RunTest.java

```diff
@@ -0,0 +1,468 @@
+/*
+ * Copyright (C) 2019 Brett Wooldridge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.zaxxer.nuprocess;
+
+import com.zaxxer.nuprocess.codec.NuAbstractCharsetHandler;
+import org.junit.*;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.CoderResult;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.zip.Adler32;
+
+/**
+ * Performs <i>synchronous</i> tests using {@link NuProcessBuilder#run}.
+ */
+public class RunTest
+{
+   private String command;
+
+   @Rule
+   public final TemporaryFolder tmp = new TemporaryFolder();
+
+   @Before
+   public void setup()
+   {
+      command = "cat";
+      if (System.getProperty("os.name").toLowerCase().contains("win")) {
+         command = "src\\test\\java\\com\\zaxxer\\nuprocess\\cat.exe";
+      }
+   }
+
+   @Test
+   public void lotOfProcesses() throws InterruptedException
```

Comment:
This test is a little different from `CatTest.lotOfProcesses()`. That test kicks off 100 processes, which get portioned across the background threads and then it waits for them all to complete. It does that 20 times. (2000 processes total)

This test spins up 2-4 daemon threads and each thread runs 20 processes _serially_ and then returns. (40-80 processes total) I could do 100 processes serially on each thread, but I wasn't sure there was a lot of value in the extra invocations.